JS/register.js

```
@@ -0,0 +1,46 @@
+const API_URL_base = "https://api.noroff.dev/api/v1";
+const registerNewUserUrl = `${API_URL_base}/social/auth/register`;
+
+async function registerNewUser(url, userData) {
+  try {
+    const registerPostData = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(userData),
+    };
+    const response = await fetch(url, registerPostData);
+
+    if (response.ok) {
+      // Registration was successful
+      window.location.href = "../index.html?registration=success";
+    } else if (response.status === 400){
+      const errorMessageElement = document.getElementById("errorMessage");
+      errorMessageElement.textContent = "User already exists, please login.";
+      errorMessageElement.style.display = "block";
+    }
+  } catch (error) {
+    
```

Do you want to return the error?

---

yes, error is now being returned